### PR TITLE
Add node e2e tests for hostIPC

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

Add node e2e tests for hostIPC.

**Which issue this PR fixes**

Part of #44118.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/assign @Random-Liu @yujuhong

### CODE REVIEW

**Comment by feiskyer** [comment]
ping @Random-Liu @yujuhong

File: test/e2e_node/security_context_test.go [hunk 1]
```diff
@@ -114,4 +115,74 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+			))
+
+			podClient.WaitForSuccess(podName, framework.PodStartTimeout)
+		}
+
+		hostSharedMemoryID := ""
+		BeforeEach(func() {
+			output, err := exec.Command("sh", "-c", "ipcmk -M 1M | awk '{print $NF}'").Output()
+			if err != nil {
+				framework.Failf("Create shared memory on the host failed: %v", err)
```
Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: "Failed to create the shared memory on the host: %v"

File: test/e2e_node/security_context_test.go [hunk 2]
```diff
@@ -114,4 +115,74 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+				[]string{"sh", "-c", "ipcs -m | awk '{print $2}'"},
+				hostNetwork,
+			))
+
+			podClient.WaitForSuccess(podName, framework.PodStartTimeout)
+		}
+
+		hostSharedMemoryID := ""
+		BeforeEach(func() {
+			output, err := exec.Command("sh", "-c", "ipcmk -M 1M | awk '{print $NF}'").Output()
```
Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: Clean up the resource after the test.

**Review by yujuhong — COMMENTED** [review]
Looks good overall with some nits.

File: test/e2e_node/security_context_test.go (hunk 2 above)
Comment (feiskyer) [inline test/e2e_node/security_context_test.go]: ack

File: test/e2e_node/security_context_test.go (hunk 1 above)
Comment (feiskyer) [inline test/e2e_node/security_context_test.go]: ack

**Comment by feiskyer** [comment]
@yujuhong Addressed comments. PTAL.

**Comment by yujuhong** [comment]
/lgtm

@Random-Liu ping for approval

**Pull request merged**